internal/controller/daemonset/instaslice_daemonset.go

```diff
@@ -151,6 +152,23 @@ func (r *InstaSliceDaemonsetReconciler) Reconcile(ctx context.Context, req ctrl.
 		return ctrl.Result{}, nil
 	}
 
+	// get the node object to watch out for any related events
+	var node v1.Node
+	if err = r.Get(ctx, client.ObjectKey{Name: nodeName}, &node); err != nil {
+		log.Error(err, "error getting the node object with ", "name", nodeName)
+		return ctrl.Result{RequeueAfter: controller.Requeue1sDelay}, nil
+	}
+	for _, cond := range node.Status.Conditions {
+		if cond.Type == v1.NodeReady && cond.Status == v1.ConditionFalse {
```

Comment:
A node can go NotReady and not be rebooted... We might have to look at the bootid or change the logic to have the daemonset reset the instaslice at daemonset startup.

Reply:
I know bootId can be fetched from the Node Status. How can we determine if it is changed? Do we have to persist boot ids somewhere?